File: safety_controller/safety_controller.py
import rclpy
from rclpy.node import Node
from sensor_msgs.msg import Joy
from geometry_msgs.msg import TwistStamped

class JoyTwistMux(Node):
    def __init__(self):
        super().__init__('joy_twist_mux')
        
        self.auto_cmd_sub = self.create_subscription(
            TwistStamped, '/j100_0000/autonomous/cmd_vel', self.auto_cmd_callback, 10)
        self.joy_sub = self.create_subscription(
            Joy, '/j100_0000/joy_teleop/joy', self.joy_callback, 10)
        
        self.auto_mode_pub = self.create_publisher(TwistStamped, '/j100_0000/auto_mode/cmd_vel', 10)

        self.auto_cmd = TwistStamped()

    def auto_cmd_callback(self, msg: TwistStamped):
        self.auto_cmd = msg

    def joy_callback(self, msg: Joy):
        axis_value = msg.axes[4]

        if axis_value == 1.0:
            self.auto_mode_pub.publish(self.auto_cmd)

def main(args=None):
    rclpy.init(args=args)
    node = JoyTwistMux()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/safety_controller.cpp
#include <rclcpp/rclcpp.hpp>
#include <geometry_msgs/msg/twist_stamped.hpp>
#include <sensor_msgs/msg/joy.hpp>

using namespace std;

class SafetyController: public rclcpp::Node{
public:
    SafetyController(): Node("safety_controller"){
        auto_cmd_vel_subscriber_ = this->create_subscription<geometry_msgs::msg::TwistStamped>(
            "autonomous/cmd_vel", 1, bind(&SafetyController::auto_cmd_vel_callback, this, placeholders::_1));
        joy_subscriber_ = this->create_subscription<sensor_msgs::msg::Joy>(
            "joy_teleop/joy", 1, bind(&SafetyController::joy_callback, this, placeholders::_1));

        auto_mode_cmd_vel_publisher_ = this->create_publisher<geometry_msgs::msg::TwistStamped>(
            "auto_mode/cmd_vel", 10);
    }

private:
    void auto_cmd_vel_callback(geometry_msgs::msg::TwistStamped::SharedPtr msg){
        auto_cmd_vel_ = msg;
    }

    void joy_callback(sensor_msgs::msg::Joy::SharedPtr msg){
        float axis_value = msg->axes[4];

        if (axis_value == 1.0){
            if (!auto_cmd_vel_) {
                RCLCPP_WARN(this->get_logger(), "Haven't received Autonomous Command Yet");
                return;
            }
            
            auto_mode_cmd_vel_publisher_->publish(*auto_cmd_vel_);
        }
    }

    rclcpp::Subscription<geometry_msgs::msg::TwistStamped>::SharedPtr auto_cmd_vel_subscriber_;
    rclcpp::Subscription<sensor_msgs::msg::Joy>::SharedPtr joy_subscriber_;

    rclcpp::Publisher<geometry_msgs::msg::TwistStamped>::SharedPtr auto_mode_cmd_vel_publisher_;

    geometry_msgs::msg::TwistStamped::SharedPtr auto_cmd_vel_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    shared_ptr<SafetyController> node = make_shared<SafetyController>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
